### DISCUSSION TITLE
Possibility to load images synchronously from ByteArray

### DISCUSSION THREAD

**Discussion by htmiel** [airsdk/Adobe-Runtime-Support#1664] [Ideas]
Currently, when you have the bytes of an image file, Loader.loadBytes() will work async and finalize loading the bitmap in the next 1-2 frames.

I know there are AS3 libraries for decoding PNG/JPG, but they are slow for large images (4k).

It would be of great use if there was an option load an image from a ByteArray and decode it synchronously.

**Comment by ajwfrost** [comment]
That should be relatively straightforward (I'd also wanted this capability in the past!) - there are internal APIs that we have available for synchronous JPEG/PNG decoding so we can expose these via a new AS3 API....

Thanks for the suggestion :-)

**Reply by toysonlocation** [reply]
++ ++ would be great for little bits
